go/libkb/interfaces.go

```diff
@@ -423,3 +423,15 @@ type Clock interface {
 type GregorDismisser interface {
 	DismissItem(id gregor.MsgID) error
 }
+
+type GregorInBandMessageHandler interface {
+	IsAlive() bool
+	Name() string
+	HandlesCategory(category string) bool
```

Comment:
I'm wondering whether it would be cleaner to take `HandlesCategory` out of the interface here and just let implementations take care of it inside of `Create` and `Dismiss`. I can see a couple problems coming up:

- Implementations that handle more than one message category will need to duplicate their category switching in both `HandlesCategory` and in `Create`/`Dismiss`.
- Implementations that filter messages in a more complicated way than just by category name (maybe by the combination of category and some other fields in the body), won't be able to rely on `HandlesCategory` at all.